addons/mass_mailing/models/mailing.py

```diff
@@ -426,25 +426,31 @@ def action_view_clicked(self):
         }
 
     def action_view_opened(self):
-        return self._action_view_documents_filtered('opened')
+        return self._action_view_documents_filtered('open')
 
     def action_view_replied(self):
-        return self._action_view_documents_filtered('replied')
+        return self._action_view_documents_filtered('reply')
 
     def action_view_bounced(self):
-        return self._action_view_documents_filtered('bounced')
+        return self._action_view_documents_filtered('bounce')
 
     def action_view_delivered(self):
         return self._action_view_documents_filtered('delivered')
 
     def _action_view_documents_filtered(self, view_filter):
-        if view_filter in ('opened', 'replied', 'bounced'):
-            opened_stats = self.mailing_trace_ids.filtered(lambda stat: stat[view_filter])
-        elif view_filter == ('delivered'):
-            opened_stats = self.mailing_trace_ids.filtered(lambda stat: stat.sent and not stat.bounced)
+        if view_filter in ('reply', 'bounce'):
+            found_traces = self.mailing_trace_ids.filtered(lambda trace: trace.trace_status == view_filter)
+        elif view_filter == 'open':
+            found_traces = self.mailing_trace_ids.filtered(lambda trace: trace.trace_status in ('open', 'reply'))
+        elif view_filter == 'click':
+            found_traces = self.mailing_trace_ids.filtered(lambda trace: trace.links_click_done)
+        elif view_filter == 'delivered':
+            found_traces = self.mailing_trace_ids.filtered(lambda trace: trace.trace_status in ('sent', 'open', 'reply'))
+        elif view_filter == 'sent':
+            found_traces = self.mailing_trace_ids.filtered(lambda trace: trace.sent_datetime)
```

Comment:
There is a bit of mismatch between all of those, they should be either 
- all verbs (reply, open, bounce, deliver, send)
- all past tense (replied, opened, bounced, delivered, sent)